frontend/src/app/services/game.ts

```diff
@@ -188,32 +190,42 @@ export class Game {
       }
 
       // mitigation not set for the level
-      if (mitigations[mitigationParam.id] !== mitigationParam.level) return;
+      if (this.mitigations[mitigationParam.id] !== mitigationParam.level) return;
 
-      this.applyMitigationEffect(ret, mitigationParam);
+      Game.applyMitigationEffect(ret, mitigationParam);
       bordersClosed ||= mitigationParam.flags.isBorders;
     });
 
     ret.exposedDrift += bordersClosed ? this.infectionsWhenBordersClosed : this.infectionsWhenBordersOpen;
 
-    eventMitigations.forEach(em => this.applyMitigationEffect(ret, em));
+    this.eventMitigations.forEach(em => Game.applyMitigationEffect(ret, em));
+
+    // All new mitigations applied this day are stored in mitigation history
+    const newEventMitigations = this.mitigationHistory[date]?.eventMitigations;
+    if (newEventMitigations) {
+      newEventMitigations
+        .filter(e => e.oneTimeEffect !== undefined)
+        .forEach(e => Game.applyMitigationEffect(ret, e));
+    }
 
     return ret;
   }
 
-  private applyMitigationEffect(affected: MitigationEffect, applied: MitigationEffect) {
-      affected.rMult *= applied.rMult;
-      affected.exposedDrift += applied.exposedDrift;
-      affected.cost += applied.cost;
-      affected.stabilityCost += applied.stabilityCost;
-      affected.vaccinationPerDay += applied.vaccinationPerDay;
+  private static applyMitigationEffect(affected: MitigationEffect, applied: Partial<MitigationEffect>) {
```

Comment:
This is not a clean approach. We are also feeding the function with "extensions" of `MitigationEffect` (`EventMitigation`) that contain additional keys.

I would not be afraid to be explicit (name keys for multiplication and name keys for addition). Type check will do  the job for us.